gems/sorbet-runtime/test/types/edge_cases.rb

```diff
@@ -15,6 +15,20 @@ def ==(other)
     assert_equal(klass.new, klass.new)
   end
 
+  it 'handles attr_writer and attr_accessor generated writers' do
+    klass = Class.new do
+      extend T::Sig
+      extend T::Helpers
+      sig { params(foo: String).returns(String) }
+      attr_writer :foo
+      sig { params(bar: Integer).returns(Integer) }
+      attr_accessor :bar
```

Comment:
Note that the `attr_reader` triggers the `method_defined` hook first, and the second method is not given runtime type checks.

Can you write a test that calls `foo = ...` with an invalid type argument so that we can see the behavior? I expect that the `attr_writer`-defined method will raise a TypeError, and the `attr_accessor`-defined method will not.

Reply:
Hmm, let me try that.

Reply:
I have no preference whether you want to handle the fact that those two have different runtime behavior in this PR, in another PR, or just leave it written in an issue somewhere.

Reply:
Indeed that is the case. I was surprised by this though. Is this always the case for runtime checker, that it would not handle an `attr_accessor` after a `attr_writer`? Or is this a test specific thing?

I will split the two tests!

Reply:
Oh, wait, I think I see the problem now. None of the `attr_accessor` writer methods are ever type-checked! I see what you mean by the reader triggering the `method_defined` hook first.

In that case, I think I'll leave the test cases as is, since if we ever find a way to allow `attr_accessor` writers to be type-checked, the test will automatically start working as-is.

Do we have an issue for the `attr_accessor` writers not being type-checked at runtime?